src/Symfony/Component/Security/Http/Firewall/ChannelListener.php

```diff
@@ -47,6 +47,9 @@ public function handle(GetResponseEvent $event)
         if ('https' === $channel && !$request->isSecure()) {
             if (null !== $this->logger) {
                 $this->logger->info('Redirecting to HTTPS.');
+                if ('https' === $request->headers->get('X-Forwarded-Proto')) {
```

Comment:
this could also handle rfc7239's "proto" parameter

Reply:
I've added an additional check for this parameter.